frontend/components/language/Selector.vue

```diff
@@ -0,0 +1,30 @@
+<script setup lang="ts">
+type GenericLang = 'en' | 'de' | 'fr' | 'es'
+interface Option { label: string, value: GenericLang }
+
+interface Props {
+  value: GenericLang
+}
+const props = defineProps<Props>()
+const emits = defineEmits<(e: 'update:value', value: GenericLang) => void>()
+const value = computed<GenericLang>({
+  get: () => props.value,
+  set: (v: GenericLang) => { emits('update:value', v) },
+})
+
+const options: Option[] = [
+  { value: 'en', label: '🇬🇧 English' },
+  { value: 'fr', label: '🇫🇷 Français' },
+  { value: 'de', label: '🇩🇪 Deutsch' },
+  { value: 'es', label: '🇪🇸 Español' },
```

Comment:
optional: instead of putting the emoji in the label and trusting the user's browser to emojify it correctly, you could use a dropdown template. In fact, [the PrimeVue example for custom templates uses flags](https://primevue.org/dropdown/#template). They do this crazy thing where they use [a single, loooong image](https://primefaces.org/cdn/primevue/images/flag/flags_responsive.png) with all the flags and then basically index into it with `background-image`.

uhh, or emojis in the label might be fine too :shrug:

Reply:
https://answers.microsoft.com/en-us/windows/forum/all/flag-emoji/85b163bc-786a-4918-9042-763ccf4b6c05?page=4#:~:text=LUDICROUS%2DREDICULOUS%2DDISGUSTING%2DPATHETIC

Yep we'll use images. Done.